pom.xml

```diff
@@ -98,8 +98,6 @@
     <fasterxml.jackson.databind.version>${fasterxml.spark3.version}</fasterxml.jackson.databind.version>
     <fasterxml.jackson.module.scala.version>${fasterxml.spark3.version}</fasterxml.jackson.module.scala.version>
     <fasterxml.jackson.dataformat.yaml.version>${fasterxml.spark3.version}</fasterxml.jackson.dataformat.yaml.version>
-    <!-- NOTE: By default we're not including JSR310 module for Jackson since Spark >= 3.x is bundling it anyway -->
-    <fasterxml.jackson.datatype.jsr310.module/>
```

Comment:
why are removing it from main pom? By removing, this module will be included which does not seem to be the intended behavior as per the comment.

Reply:
It seems like the original intension was to not shade JSR310 module when spark version >= 3.3. All other jackson packages that are used are already being shaded. It didn't look like this special treatement for JSR310 package was needed, so this line is being removed to shade JSR310 just like any other jackson package that is being used.

Reply:
> this special treatement for JSR310 package was needed

can you help clarify why? the intention was to avoid duplicated bundling of this module.

Reply:
@xushiyan Not sure if I understand what you mean by "duplicated bundling" here (?). Since JSR310 package is being shaded and included in the bundle, hudi code will go to this shaded package instead of another JSR310 package that may be present in the classpath at runtime. From what I can see other jackson packages are also similarly shaded.

Reply:
@amrishlal You are right, if Hudi bundle jar shade the class anyway, we should always include the jar in the bundle, or any reference to the JSR class could encounter class not found exception.

Another choice is we do not shade the JSR clazz, do we need a shade here?

Reply:
@danny0405 For now, we seem to be shading all the jackson packages and hence it seems like a good idea to shade jackson JSR package as well to keep the versions across jackson packages consistent (otherwise testing may become problematic). But yes, as a future item we could look into not shading any of the jackson packages at all for spark versions (>= 3.1 I think) where Jackson is already being included. We would also need to confirm on the Flink side (as I was seeing some errors in Flink validate bundles tests). For now though we would like to get our integration test setup up and running as that is blocked by this PR.

Reply:
as this variable is removed, can you global search and make sure all refs are updated?

Reply:
@xushiyan Fixed. Also verified that the correct version of JSR package (matching what is including in spark 3.x runtime) is being included.